### PR TITLE
#1713 Bill of Materials Frontend

### LINKED ISSUE #1511: [BOM] - Update Project Transformer to Return BOM Data
### Description

Blocked By #1479 

With this new BOM feature, we've added added a list of assemblies and materials to the project object in the database. We need to update the project transformer and project query args in the backend so that the new assemblies and materials along with their associated objects are correctly returned to the frontend.

### Acceptance Criteria

- [ ] the wbsElement in the project query args has been configured to also fetch the assemblies and materials associated with it
- [ ] query args/ transformers have been added for Material and Assembly (should be used in the above bullet point)
- [ ] in the projects transformer these are returned as assemblies and materials elements on the project
- [ ] the wbsElement shared type is updated to include an array of assemblies and materials

### Proposed Solution

You might need to create query args/ transformers for Manufacturer and MaterialType if they have not been created in the getAll endpoints for each of them.

### Mocks

_No response_

### LINKED ISSUE #1713: Add Material Form
### Description

We want users to be able to add materials to a bill of materials on a project. In this ticket, you're gonna create the form in a modal that pops up to facilitate that

### Acceptance Criteria

- Should be abstracted so that an edit or create endpoint can be called on submit
- Should have all the forms in the mock below
- Should call the appropriate endpoints to get the options for materials and manufacturers
- direct any other question to Reid/ Mihir in slack

### Proposed Solution

Make dah form

### Mocks

image

### PR DESCRIPTION
**Pull request by Peyton-McKee**
## Changes

All the frontend code for bill of materials! See #1427 for reference on all the features/mocks

## To Do

- [x] add create assembly
- [x] add assign to assembly
- [x] make total on the bottom a select
- [x] add delete
- [x] create suggested budget increase CR
- [x] make statuses pretty
- [x] make buttons on the bottom pretty

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #1713  (issue #)
Closes #1511

### CODE REVIEW

File: src/shared/src/types/bom-types.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+export type AssemblyPreview = Omit<Assembly, 'materials'>;
+
+export interface Manufacturer {
+  name: string;
+  dateCreated: Date;
+  creatorId: number;
+  creator: User;
+  materials: MaterialPreview[];
```
Comment (RChandler234) [inline src/shared/src/types/bom-types.ts:46]: a good chunk of these types exist in projects-types rn fyi, we'll need to make sure every place that was using those uses these now, and that the shared types actually make sense

Comment (Peyton-McKee) [reply src/shared/src/types/bom-types.ts:48]: Yesh i deleted them

**Comment by RChandler234** [comment]
current look:
image

**Comment by RChandler234** [comment]
image

**Comment by RChandler234** [comment]
also switched up the tab order. I think this is more inline with how much these tabs actually get used/ how much effort we've put into making them useful:
image

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥

**Pull request merged**